tv_apps/fling-player/js/connector.js

```diff
@@ -1,127 +1,194 @@
-      this.initSession(presentation.session);
+    if (this._presentation.session) {
+      console.log('init session');
+      this._initSession(this._presentation.session);
     } else {
-      presentation.addEventListener('sessionready', this);
+      console.log('no session, so listent to sessionready event.');
+      this._presentation.addEventListener('sessionready', this);
     }
   };
 
-  proto.initSession = function c_initSession(session) {
-    if (!presentation) {
-      return;
+  proto._initSession = function c_initSession(session) {
+
+    console.log('Connector#_initSession');
+    if (!this._presentation) {
+      throw new Error('Init session without the presentation object.');
     }
 
     this._session = session;
-    this._session.onmessage = this.handleEvent.bind(this);
-    this.initEvents();
-  };
+    console.log('this._session = ', this._session);
 
-  proto.initEvents = function c_initEvents() {
-    if (!this._player) {
-      console.error('Connector doesn\'t have player object.');
-      return;
-    }
+    if (this._session.state !== 'connected') {
 
-    this._player.on('loaded', this.reportStatus.bind(this, 'loaded'));
-    this._player.on('seeked', this.reportStatus.bind(this, 'seeked'));
-    this._player.on('playing', this.reportStatus.bind(this, 'playing'));
-    this._player.on('timeupdate', this.reportStatus.bind(this, 'timeupdate'));
-    this._player.on('stopped', this.reportStatus.bind(this, 'stopped'));
-    this._player.on('buffering', this.reportStatus.bind(this, 'buffering'));
-    this._player.on('buffered', this.reportStatus.bind(this, 'buffered'));
-    this._player.on('error', this.reportStatus.bind(this, 'error'));
-  };
+      console.log('not connected state so listen to onstatechange');
 
-  proto.handleRemoteMessage = function c_handleRemoteMessage(msg) {
-    // We don't process the out of dated message.
-    if (this._lastReceived >= msg.seq) {
-      this.replyACK(msg, 'wrong-seq');
-      return;
-    }
+      this._session.onstatechange = function() {
 
-    this._lastReceived = msg.seq;
-    if (!this._player) {
-      console.error('Connector doesn\'t have player object.');
-      this.replyACK(msg, 'player-error');
-      return;
-    }
-    var err;
-    switch(msg.type) {
-      case 'load':
-        this._player.load(msg.url);
-        break;
-      case 'play':
-        this._player.play();
-        break;
-      case 'pause':
-        this._player.pause();
-        break;
-      case 'seek':
-        try {
-          this._player.seek(msg.time);
-        } catch(e) {
-          err = e.message;
-        }
-        break;
-      default:
-        return;
+        console.log('this._session.onstatechange and state = ' +
+          this._session.state);
+
+          if (this._session.state === 'connected') {
+            this._initSession(this._session);
+          }
+      }.bind(this);
+
+    } else {
+      console.log('connected state so listen to onmessage');
```

Comment:
If you want to leave all these console.log, please set a DEBUG flag and let them be print out only when DEBUG is true.